source/posix/socket.c

```diff
@@ -1029,30 +1030,38 @@ static int s_process_write_requests(struct aws_socket *socket, bool spawned_from
     }
 
     if (purge) {
-        aws_raise_error(aws_error);
         while (!aws_linked_list_empty(&socket_impl->write_queue)) {
             struct aws_linked_list_node *node = aws_linked_list_pop_front(&socket_impl->write_queue);
             struct write_request *write_request = AWS_CONTAINER_OF(node, struct write_request, node);
 
-            write_request->written_fn(socket, aws_error, 0, write_request->write_user_data);
+            /* If aws_socket_write() invoked this fn, don't invoke its write_request's error callback.
+               The user will learn about the error from aws_socket_write()'s return value */
```

Comment:
This was hard to parse, perhaps:
If this fn was invoked directly from aws_socket_write(), don't invoke the error callback as the user will be able to rely on the return value from aws_socket_write()

Reply:
copy/pasted